[Peterborough] WasteWorks Phase 2 improvements

davea opened the pull request:
This PR primarily covers improvements to the "request new container" functionality as requested by Peterborough. There are also improvements to the "report a problem with container" flow, mainly allowing multiple problems to be reported at once.

Commits roughly align to items on [this Basecamp to-do list](https://3.basecamp.com/4020879/buckets/19439943/todolists/4020202167).

[skip changelog]

dracos commented on templates/web/base/waste/bin_days.html:
```diff
@@ -108,19 +108,21 @@ <h3>Download your collection schedule</h3>
+              [% IF show_garden_subscribe %]
+                <li><a href="[% c.uri_for_action('waste/garden', [ property.id ]) %]">Subscribe to Green Garden Waste collection</a></li>
+              [% END %]
+              [% TRY %][% PROCESS waste/_more_services_extra.html %][% CATCH file %][% END %]
```
This was only used by Peterborough so I guess you could drop it here and in Peterborough sidebar.html and include its contents directly in sidebar.html?

dracos reviewed (changes requested):
Looks good, a number of comments/queries, nothing major. Would have been good to review in smaller bits perhaps!

davea commented on perllib/FixMyStreet/App/Form/Waste/Request/Peterborough.pm:
```diff
@@ -24,4 +24,8 @@ has_field submit => (
     order => 999,
 );
 
+sub default_submit {
```
It's a magic method that FormHandler users to get the value for the `submit` field. Covered here https://github.com/mysociety/fixmystreet/blob/ac61291f9084f1069a1d55d2b1a44ebc9ebc7933/t/app/controller/waste_peterborough.t#L287

dracos replied:
Aha, ta.

davea commented on perllib/FixMyStreet/Cobrand/Peterborough.pm:
```diff
@@ -908,6 +908,7 @@ sub waste_munge_request_data {
     $data->{title} = "Request new $container";
     $data->{detail} = "Quantity: $quantity\n\n$address";
     $data->{detail} .= "\n\nReason: $reason" if $reason;
+    $data->{detail} .= "\n\nPlease remove cracked bin." if $reason && $reason eq "Cracked bin";
```
🤦 Good spot!

davea commented on perllib/Integrations/Bartec.pm:
```diff
@@ -212,7 +212,7 @@ sub Jobs_FeatureScheduleDates_Get {
 
     $start = $w3c->format_datetime($start || DateTime->now->subtract(days => $days_buffer));
-    $end = $w3c->format_datetime($end || DateTime->now);
+    $end = $w3c->format_datetime($end || DateTime->now->add(days => $days_buffer));
```
The commit message probably needs updating to explain this change too. Basically we need to search in the future too because of [an apparent Bartec bug](https://3.basecamp.com/4020879/buckets/19439943/todos/4365548265).

davea commented:
Thanks for the review. I've addressed your comments and also a couple of last-minute comments [from the client](https://3.basecamp.com/4020879/buckets/19439943/todos/4431074593#__recording_4525578108). I'll reword those commits you mention ([1](https://github.com/mysociety/fixmystreet/pull/3742#discussion_r783882052), [2](https://github.com/mysociety/fixmystreet/pull/3742#discussion_r783879150)) when I rebase ahead of merging.
